src/rosdiscover/config.py

```diff
@@ -62,14 +119,23 @@ def from_dict(cls, dict_: Mapping[str, Any]) -> 'Config':
         if has_environment and not isinstance(dict_['environment'], dict):
             raise ValueError("expected 'environment' to be a mapping")
 
+        has_node_sources = 'node_sources' in dict_
+        if has_node_sources and not isinstance(dict_['node_sources'], list):
+            raise ValueError("expected 'node_sources' to be a list")
+
         image: str = dict_['image']
-        sources: Sequence[str] = dict_['sources']
-        launches: Sequence[str] = dict_['launches']
-        environment: Mapping[str, str] = dict(dict_.get('environment', {}))
+        sources: t.Sequence[str] = dict_['sources']
+        launches: t.Sequence[str] = dict_['launches']
+        environment: t.Mapping[str, str] = dict(dict_.get('environment', {}))
+        node_sources: t.Sequence[t.Dict[str, t.Any]] = list(dict_.get('node_sources', []))
+
         return Config(image=image,
                       sources=sources,
                       launches=launches,
-                      environment=environment)
+                      environment=environment,
```

Comment:
do you mind if we split the arguments to this function call across separate lines? readability becomes a little strained by line 136.

Reply:
Done